File: python/day12.py
import json

def add(thing):
	total = 0
	if type(thing) is dict:
		for (key, item) in thing.items():
			total += add(item)
	elif type(thing) is list:
		for item in thing:
			total += add(item)
	elif type(thing) is int or (type(thing) is str and thing.isnumeric()):
		return int(thing)
	return total

fp = open('../input/input12.txt')
json = json.load(fp)
print add(json)

File: php/day12.php
<?php
function add($input)
{
	if (!is_array($input))
	{
		return $input;
	}
	$total = 0;
	foreach ($input as $key => $item)
	{
		if (is_array($item))
		{
			$total += add($item);
		}
		else
		{
			$total += $item;
		}
	}
	return $total;
}

$json = json_decode(file_get_contents('../input/input12.txt'), true);
echo add($json) . PHP_EOL;
